### PR TITLE
Remove `libcrux` dependency in `libcrux-kem`

### PR DESCRIPTION
**Pull request by jschneider-bensch**
Same procedure as #347, replacing `libcrux` `Drbg` with `OsRng`.

### CODE REVIEW

File: libcrux-kem/Cargo.toml [hunk 1]
```diff
@@ -25,5 +25,5 @@ pre-verification = ["libcrux-ml-kem/pre-verification"]
 
 [dev-dependencies]
 libcrux-kem = { version = "0.0.2-alpha.1", path = "./", features = ["tests"] }
-libcrux = { version = "0.0.2-alpha.1", path = "../", features = ["rand"] }
+rand_core = { version = "0.6" }
```
Comment (franziskuskiefer) [inline libcrux-kem/Cargo.toml:28]: No need for this if you use `rand::rngs::OsRng` instead.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]

File: libcrux-kem/Cargo.toml (hunk 1 above)
Comment (jschneider-bensch) [reply libcrux-kem/Cargo.toml]: You' right, thanks! :+1:

File: libcrux-kem/Cargo.toml [hunk 2]
```diff
@@ -25,5 +26,4 @@ pre-verification = ["libcrux-ml-kem/pre-verification"]
 
 [dev-dependencies]
 libcrux-kem = { version = "0.0.2-alpha.1", path = "./", features = ["tests"] }
-libcrux = { version = "0.0.2-alpha.1", path = "../", features = ["rand"] }
 hex = { version = "0.4.3", features = ["serde"] }
```
Comment (franziskuskiefer) [inline libcrux-kem/Cargo.toml:29]: There shouldn't be a version here. That prevents it from being released. It should always take the current version.

**Review by franziskuskiefer — APPROVED** [review]
Thanks!

**Pull request merged**